### ISSUE TITLE
Let's document: aws

### ISSUE THREAD

**Issue by Waples** [tldr-pages/tldr#5653]
- [ ] accessanalyzer →
- [ ] acm →
- [x] acm-pca → #13327
- [ ] alexaforbusiness →
- [ ] amplify →
- [ ] amplifybackend →
- [ ] apigateway →
- [ ] apigatewaymanagementapi →
- [ ] apigatewayv2 →
- [ ] appconfig →
- [ ] appflow →
- [ ] appintegrations →
- [ ] application-autoscaling →
- [ ] application-insights →
- [ ] appmesh →
- [ ] appstream →
- [ ] appsync →
- [ ] athena →
- [ ] auditmanager →
- [ ] autoscaling →
- [ ] autoscaling-plans →
- [x] backup → #9369
- [x] batch → #10570, #10591
- [ ] braket →
- [ ] budgets →
- [x] ce → #13354
- [ ] chime →
- [x] cloud9 → #10627
- [ ] clouddirectory
- [x] cloudformation → #10116
- [ ] cloudfront →
- [ ] cloudhsm →
- [ ] cloudhsmv2 →
- [ ] cloudsearch →
- [ ] cloudsearchdomain →
- [ ] cloudtrail →
- [x] cloudwatch → #10614
- [x] codeartifact → #11224
- [ ] codebuild →
- [x] codecommit → #9202
- [ ] codeguru-reviewer →
- [ ] codeguruprofiler →
- [ ] codepipeline →
- [ ] codestar →
- [ ] codestar-connections →
- [ ] codestar-notifications →
- [ ] cognito-identity →
- [x] cognito-idp → #9032
- [ ] cognito-sync →
- [ ] comprehend →
- [ ] comprehendmedical →
- [ ] compute-optimizer →
- [ ] configservice →
- [x] configure → #8808
- [ ] connect →
- [ ] connect-contact-lens →
- [ ] connectparticipant →
- [x] cur → #6133
- [ ] customer-profiles →
- [ ] databrew →
- [ ] dataexchange →
- [ ] datapipeline →
- [ ] datasync →
- [ ] dax →
- [ ] deploy →
- [ ] detective →
- [ ] devicefarm →
- [ ] devops-guru →
- [ ] directconnect →
- [ ] discovery →
- [ ] dlm →
- [ ] dms →
- [ ] docdb →
- [ ] ds →
- [x] dynamodb → #11570
- [ ] dynamodbstreams →
- [ ] ebs →
- [x] ec2 →#4661
- [ ] ec2-instance-connect →
- [x] ecr → #6134
- [ ] ecr-public →
- [x] ecs → #4661
- [ ] efs →
- [x] eks → #11577
- [ ] elastic-inference →
- [ ] elasticache →
- [ ] elasticbeanstalk →
- [ ] elastictranscoder →
- [ ] elb →
- [ ] elbv2 →
- [ ] emr →
- [ ] emr-containers →
- [ ] es →
- [ ] events →
- [ ] firehose →
- [ ] fms →
- [ ] forecast →
- [ ] forecastquery →
- [ ] frauddetector →
- [ ] fsx →
- [ ] gamelift →
- [ ] glacier →
- [ ] globalaccelerator →
- [x] glue → #4740
- [ ] greengrass →
- [ ] groundstation →
- [ ] guardduty →
- [ ] health →
- [ ] healthlake →
- [x] help → #6446 
- [x] history → #8805
- [ ] honeycode →
- [x] iam → #4662 
- [ ] identitystore →
- [ ] imagebuilder →
- [ ] importexport →
- [ ] inspector →
- [ ] iot →
- [ ] iot-data →
- [ ] iot-jobs-data →
- [ ] iot1click-devices →
- [ ] iot1click-projects →
- [ ] iotanalytics →
- [ ] iotevents →
- [ ] iotevents-data →
- [ ] iotsecuretunneling →
- [ ] iotsitewise →
- [ ] iotthingsgraph →
- [ ] ivs →
- [x] kafka → #11696
- [x] kendra → #13315
- [x] kinesis → #4510
- [ ] kinesis-video-archived-media →
- [ ] kinesis-video-media →
- [ ] kinesis-video-signaling →
- [ ] kinesisanalytics →
- [ ] kinesisanalyticsv2 →
- [ ] kinesisvideo →
- [ ] kms →
- [ ] lakeformation →
- [x] lambda → #4774
- [ ] lex-models
- [ ] lex-runtime
- [ ] license-manager
- [x] lightsail → #8934
- [ ] logs →
- [ ] lookoutvision →
- [ ] machinelearning →
- [ ] macie →
- [ ] macie2 →
- [ ] managedblockchain →
- [ ] marketplace-catalog →
- [ ] marketplace-entitlement →
- [ ] marketplacecommerceanalytics →
- [ ] mediaconnect →
- [ ] mediaconvert →
- [ ] medialive →
- [ ] mediapackage →
- [ ] mediapackage-vod →
- [ ] mediastore →
- [ ] mediastore-data →
- [ ] mediatailor →
- [ ] meteringmarketplace →
- [ ] mgh →
- [ ] migrationhub-config →
- [ ] mobile →
- [ ] mq →
- [ ] mturk →
- [ ] mwaa →
- [ ] neptune →
- [ ] network-firewall →
- [ ] networkmanager →
- [ ] opsworks →
- [ ] opsworks-cm →
- [ ] organizations →
- [ ] outposts →
- [ ] personalize →
- [ ] personalize-events →
- [ ] personalize-runtime →
- [ ] pi →
- [ ] pinpoint →
- [ ] pinpoint-email →
- [ ] pinpoint-sms-voice →
- [ ] polly →
- [x] pricing → #8804
- [ ] qldb →
- [ ] qldb-session →
- [x] quicksight → #4738
- [ ] ram →
- [x] rds → #6547
- [ ] rds-data →
- [ ] redshift →
- [ ] redshift-data →
- [ ] rekognition →
- [ ] resource-groups →
- [ ] resourcegroupstaggingapi →
- [ ] robomaker →
- [x] route53 → #8606 
- [ ] route53domains →
- [ ] route53resolver →
- [x] s3 (This command has a separate issue page #8406 to track it's subcommands) 
- [x] s3api → #6132
- [ ] s3control →
- [ ] s3outposts →
- [ ] sagemaker →
- [ ] sagemaker-a2i-runtime →
- [ ] sagemaker-edge →
- [ ] sagemaker-featurestore-runtime →
- [ ] sagemaker-runtime →
- [ ] savingsplans →
- [ ] schemas →
- [ ] sdb →
- [x] secretsmanager → #6131
- [ ] securityhub →
- [ ] serverlessrepo →
- [ ] service-quotas →
- [ ] servicecatalog →
- [ ] servicecatalog-appregistry →
- [ ] servicediscovery →
- [x] ses → #7136
- [ ] sesv2 →
- [ ] shield →
- [ ] signer →
- [ ] sms →
- [ ] snowball →
- [x] sns → #12259
- [x] sqs → #9085
- [ ] ssm →
- [x] sso → #14115
- [ ] sso-admin →
- [ ] sso-oidc →
- [ ] stepfunctions →
- [ ] storagegateway →
- [x] sts → #8669
- [ ] support →
- [ ] swf →
- [ ] synthetics →
- [ ] textract →
- [ ] timestream-query →
- [ ] timestream-write →
- [ ] transcribe →
- [ ] transfer →
- [ ] translate →
- [ ] waf →
- [ ] waf-regional →
- [ ] wafv2 →
- [ ] workdocs →
- [ ] worklink →
- [x] workmail  → #9058
- [ ] workmailmessageflow →
- [ ] workspaces →
- [ ] xray →

**Comment by Waples** [comment]
@bl-ue whenever you are bored :P #HelpWanted :p

**Comment by bl-ue** [comment]
Haha @Waples did you happen to see this: #5212?

**Comment by bl-ue** [comment]
@Waples it looks like you might have removed all occurences of the letter `o` in the above command list 😄, e.g. `alexafrbusiness`, `wrkmailmessageflw`, `applicatin-autscaling`

**Comment by Waples** [comment]
> @Waples it looks like you might have removed all occurences of the letter `o` in the above command list , e.g. `alexafrbusiness`, `wrkmailmessageflw`, `applicatin-autscaling`

Owh haha, think i was a bit too agressive with my `%s/\so^/g` ish replace then xD will fix that in a second

**Comment by Waples** [comment]
Fixed it =D

**Comment by Waples** [comment]
@bl-ue didn't see that issue before, but that goes in a bit too deep with ALL the commands. 
I was thinking more like, for example [aws-ec2.md](https://github.com/tldr-pages/tldr/blob/master/pages/common/aws-ec2.md).

**Comment by bl-ue** [comment]
You're right, that's why I closed it.  But I wanted to link it for reference 👍🏻

**Comment by Waples** [comment]
I'm gonna make some time today to add at least 15 examples, but I don't know what the best course of action is, a seperate branch for every page added or in one big PR ? 
@bl-ue  what ya want

**Comment by bl-ue** [comment]
The former @Waples.  Please open 1 PR per new page.  It makes it easier to review the changes.

It would certainly be great if we could document all of the `aws` subcommands—I'd say they're probably looked for often 🤷🏻

**Comment by Waples** [comment]
Organization status, here I come. xD

**Comment by Waples** [comment]
> I'd say they're probably looked for often 🤷🏻
sounds a bit sarcastic xD

**Comment by bl-ue** [comment]
Haha no I wasn't intended to be sarcastic actually :)

**Comment by sbrl** [comment]
Oh wow, that's an _enormous_ list @Waples! I'd suggest starting small, and opening just 1 or 2 PRs for 1 or 2 pages at a time to get used to the tldr page syntax if you haven't already got experience with the review process.

**Comment by 258204** [comment]
Starting one on secrets manager now.

**Comment by 258204** [comment]
Starting s3api and cur

**Comment by 258204** [comment]
Might do ecs and eks next, everyone loves container orchestration.

**Comment by bl-ue** [comment]
And do `ecr` — I'm going to upload some Docker images to ECR today, and it would be cool to know how to use `aws ecr`! 🙂

**Comment by 258204** [comment]
I will start on ecr first then.

**Comment by 258204** [comment]
It's a spiderweb. You need a tldr of docker-login for a tldr or aws ecr.

**Comment by 258204** [comment]
This actually brings up an important question.
The authoritative documentation is more complete (eg. https://awscli.amazonaws.com/v2/documentation/api/latest/reference/ecr/index.html), however, the quick start guide https://docs.aws.amazon.com/AmazonECR/latest/userguide/ is much more useful to get to know a new tool. It would deviate from the standard, but pages that include both would be more useful.

**Comment by 258204** [comment]
Also gonna add short one for docker-tag

**Comment by 258204** [comment]
And docker-image....


DOCUMENT ALL THE THINGS!!! :D

**Comment by bl-ue** [comment]
Hmm, I'm not really sure.  The first page does provide a lot more information directly related to the CLI (36 addition subcommands). The second page is tuned for beginners but isn't actually about the CLI.

I'm probably inclined to keep the first link, because it seems like a user who goes to that link might be looking for more info on how to use the tool itself, but there's equal argument for both links, so I'm chasing my own tail here. 😥

**Comment by 258204** [comment]
The second link mirrors the same steps for cli and gui.

**Comment by sbrl** [comment]
I would suggest that for more information links subpages of <https://awscli.amazonaws.com/v2/documentation/api/latest/reference/ecr/index.html> (e.g. <https://awscli.amazonaws.com/v2/documentation/api/latest/reference/ecr/get-lifecycle-policy-preview.html>) are more directly relevant to each individual page.

The generic user guide looks useful, but not specifically to the `aws` subcommand being documented.

**Comment by 258204** [comment]
Thanks @sbrl, you are correct.

**Comment by 258204** [comment]
> And docker-image....
> 
> DOCUMENT ALL THE THINGS!!! :D

It's already there, my mistake, my package manager just didn't have the latest version of tldr.

**Comment by 258204** [comment]
rds #6547

**Comment by Waples** [comment]
Thanks for all your work @258204 !
I've been very busy (work/private), hence why I didn't add anything so far to my "todo" list.
I'll take a look at the state of the AWS API tomorrow, to see if new services need to be added to the list.
I might have some time tomorrow to write some tldr pages, time will tell.

**Comment by subhash-iiitm** [comment]
aws sts #8669

**Comment by piraces** [comment]
aws pricing #8804

**Comment by piraces** [comment]
aws history #8805 8805

**Comment by piraces** [comment]
aws configure #8808

**Comment by bryanmg** [comment]
Hi! I'll work on sqs command file

**Comment by bryanmg** [comment]
aws-sqs: add page #9085

**Comment by Waples** [comment]
Thank you @piraces @bryanmg for all your work!

**Comment by Waples** [comment]
Note to self: check if the list still up to date with the v2 (and beta v3) version of aws-cli

**Comment by the0001** [comment]
aws-cloudwatch: add page #10614

**Comment by the0001** [comment]
aws-cloud9: add page #10627

**Comment by Sourav-Kumar-Panda** [comment]
Can I work on this?

**Comment by kbdharun** [comment]
> Can I work on this?

Sure

**Comment by Sourav-Kumar-Panda** [comment]
I am new to open source, could you please guide me? and what are the things I should follow in order to solve this issue

**Comment by sbrl** [comment]
Hey, @Sourav-Kumar-Panda, @nkzren! You don't need to ask - just open a pull request with a new page :-)

**Comment by Waples** [comment]
Update from myself: I might do some work with `Sagemaker` in the near future (on boto3 level), so I might document something for those items.

**Comment by Sourav-Kumar-Panda** [comment]
Hey, @Waples , please check the PR #11570

**Comment by Sourav-Kumar-Panda** [comment]
aws-eks: add page. #11577

**Comment by kbdharun** [comment]
> Should we update the pages to match the new version?

Yeah, we should also check for any changed syntax and update it (if a subcommand/flag got deprecated we should mention it is only available in v1).

**Comment by sbrl** [comment]
Don't forget to reference this PR's number in any PRs you open @sebastiaanspeck :-)

**Comment by sablokgaurav** [comment]
@Waples adding this to the aws terraform. Writing the aws terraform initator files: https://github.com/sablokgaurav/hashicorp_instance_initiator/blob/main/terraform_aws.sh

**Comment by Waples** [comment]
> > Just as an FYI, I have been ill / busy for a long while, but I do still weekly check how the process is going. Thank you all for giving you're time to fill the stupidly long list of items.
> > I will be more aware of ongoing work / MR/PRs etc from now on again, now my health has improved (beware xD).
> 
> Good to have you back! :D

thanks bruv ^^ feeling allot better now. Lets document some stuff haha

**Comment by Waples** [comment]
At the time of writing, I don't even know myself anymore if this list is complete and compatible with v3 (and v4-beta).

Note to self: investigate this.

And again, thank you all for helping with this massive project.

// Florian

**Comment by skarkloff** [comment]
Good morning, I am new to open source contributions. I saw this project and I am interested in contributing, how can I get started, whether for this issue or another?

**Comment by spageektti** [comment]
> Good morning, I am new to open source contributions. I saw this project and I am interested in contributing, how can I get started, whether for this issue or another?

Read [CONTRIBUTING.md](https://github.com/tldr-pages/tldr/blob/main/CONTRIBUTING.md) and [Style Guide](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md). This will make it easier both for you and maintainers.

If you speak lang other than English, consider translating the pages, it's the most easy way to get started.

Consider adding examples to existing pages to practice before creating new pages, but creating a new page is welcome too .

**Comment by phantompunk** [comment]
aws-sso: add page https://github.com/tldr-pages/tldr/pull/14115

**Comment by Naina-Coder123** [comment]
Hi! I’d like to work on aws-ec2 and add a TLDR page for it.

**Comment by sebastiaanspeck** [comment]
> Hi! I’d like to work on aws-ec2 and add a TLDR page for it.

Feel free to open a PR.

**Comment by Waples** [comment]
@Managor could you stop messing around with assignments?

**Comment by Managor** [comment]
I haven't done anything since you snapped at me
